src/pages-and-resources/app-settings-modal/AppSettingsModal.jsx

```diff
@@ -0,0 +1,171 @@
+        )}
+    </TransitionReplace>
+  );
+}
+
+AppSettingsForm.propTypes = {
+  formikProps: PropTypes.shape({
+    values: PropTypes.shape({ enabled: PropTypes.bool.isRequired }),
+  }).isRequired,
+  children: PropTypes.func,
+};
+
+AppSettingsForm.defaultProps = {
+  children: null,
+};
+
+function AppSettingsModal({
+  intl,
+  appId,
+  title,
+  children,
+  initialValues,
+  validationSchema,
+  onClose,
+  onSettingsSave,
+  enableAppLabel,
+  enableAppHelp,
+  learnMoreURL,
+  learnMoreText,
+}) {
+  const { courseId } = useContext(PagesAndResourcesContext);
+  const loadingStatus = useSelector(getLoadingStatus);
+  const appInfo = useModel('courseApps', appId);
+  const dispatch = useDispatch();
+  const handleFormSubmit = (values) => {
+    // If the app's enabled/disabled loadingStatus has changed, set that first.
+    if (appInfo.enabled !== values.enabled) {
+      dispatch(enableDisableApp(courseId, appInfo.id, values.enabled, true));
+    }
+    // Call the submit handler for the settings component to save its settings
+    if (onSettingsSave) {
+      onSettingsSave();
+    }
+  };
+  const learnMoreLink = (
+    <Hyperlink destination={learnMoreURL} target="_blank">{learnMoreText}</Hyperlink>
+  );
+
+  return (
+    <ModalLayer
+      isOpen
+      closeText={intl.formatMessage(messages.cancel)}
+      dialogClassName="modal-dialog-centered modal-lg"
+      onClose={onClose}
+    >
+      <div
+        role="dialog"
+        aria-label={title}
+        className="bg-white d-flex flex-column mw-xs p-3"
+      >
+        {
+          loadingStatus === RequestStatus.SUCCESSFUL && (
+            <Formik
+              initialValues={{
+                enabled: !!appInfo?.enabled,
+                ...initialValues,
+              }}
+              validationSchema={
+                Yup.object()
+                  .shape({
+                    enabled: Yup.boolean(),
+                    ...validationSchema,
+                  })
+              }
+              onSubmit={handleFormSubmit}
+            >{(formikProps) => (
+              <Form onSubmit={formikProps.handleSubmit}>
+                <h3>{title}</h3>
+                <AppConfigFormDivider />
+                <FormSwitchGroup
+                  id={`enable-${appId}-toggle`}
+                  name="enabled"
+                  onChange={formikProps.handleChange}
+                  onBlur={formikProps.handleBlur}
+                  checked={formikProps.values.enabled}
+                  label={(
+                    <>
+                      {enableAppLabel}&nbsp;
+                      <StatusBadge loadingStatus={formikProps.values.enabled} />
+                    </>
+                  )}
+                  helpText={(<p>{enableAppHelp}<br />{learnMoreLink}</p>)}
+                />
+                <AppConfigFormDivider />
+                <AppSettingsForm formikProps={formikProps}>
+                  {children}
+                </AppSettingsForm>
```

Comment:
Can you please add a little information around this? When this `AppSettingsForm` will be needed, in which app? What is the usecase for this?

Reply:
nvm -- I think these would be the settings for app other than enable/disable. Got it.